### ISSUE TITLE
CI tests with all trits encodings

### ISSUE THREAD

**Issue by thibault-martinez** [iotaledger/entangled#308]

**Comment by th0br0** [comment]
Fixed upstream in ci-pipelines via https://github.com/iotaledger/ci-pipelines/commit/4a412aef213846ec0d99f93f2e8b58f6e2cf7d8f

Wasn't enabled before because there were some bugs, but it looks like those are fixed now! Great.

**Comment by th0br0** [comment]
Partially reverting, looks like the bundle validator is having some issues: (trit_encoding=1)

```
consensus/bundle_validator/tests/bundle_validator.c:36:test_bundle_validate_empty_invalid:FAIL: Expression Evaluated To FALSE
consensus/bundle_validator/tests/bundle_validator.c:60:test_bundle_size_1_value_with_wrong_address_invalid:FAIL: Expression Evaluated To FALSE
consensus/bundle_validator/tests/bundle_validator.c:147:test_bundle_exceed_supply_pos_invalid:FAIL: Expression Evaluated To FALSE
consensus/bundle_validator/tests/bundle_validator.c:197:test_bundle_exceed_supply_neg_invalid:FAIL: Expression Evaluated To FALSE
consensus/bundle_validator/tests/bundle_validator.c:247:test_bundle_validate_size_4_value_wrong_sig_invalid:FAIL: Expression Evaluated To FALSE
consensus/bundle_validator/tests/bundle_validator.c:319:test_bundle_validate_size_4_value_valid:FAIL: Expression Evaluated To FALSE
```

Assigning to Tsvi for now. Please fix asap.

**Comment by th0br0** [comment]
bundle_validator & exit_probability_randomizer have issues with `trit_encoding=4` and `trit_encoding=5`
```
consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:163:test_single_tx_tangle:FAIL: Expression Evaluated To FALSE
consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:50:test_cw_topology_blockchain:FAIL: Expression Evaluated To FALSE
consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:50:test_cw_topology_only_direct_approvers:FAIL: Expression Evaluated To FALSE
consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:231:test_cw_topology_four_transactions_diamond:FAIL: Expression Evaluated To FALSE
consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:323:test_cw_topology_two_inequal_tips:FAIL: Expression Evaluated To FALSE
consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:448:test_1_bundle:FAIL: Expression Evaluated To FALSE
consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c:585:test_2_chained_bundles:FAIL: Expression Evaluated To FALSE
```

**Comment by thibault-martinez** [comment]
I have been investigating these errors and have a fix for most of them.
A last error seems to boil down to `flex_trits_slice` with `FLEX_TRIT_ENCODING_4_TRITS_PER_BYTE `, I might need help from @ifullgaz to understand it.

**Comment by th0br0** [comment]
Fixed via #309 and #310
